### PR TITLE
fix(extension): DApp tx signing for Trezor [LW-9071]

### PR DESCRIPTION
**Pull request by DominikGuzei**
# Checklist

- [x] https://input-output.atlassian.net/browse/LW-9071

---

## Proposed solution

This PR fixes DApp transaction signing using a Trezor hardware wallet.

### CODE REVIEW

File: apps/browser-extension-wallet/src/features/dapp/components/ConfirmTransaction.tsx [hunk 1]
```diff
@@ -166,24 +168,22 @@ export const ConfirmTransaction = withAddressBookContext((): React.ReactElement
...
-            },
-            { logger: console, runtime }
-          );
-        })
-        .catch((error) => {
-          throw error;
-        });
+      await (keyAgentType === KeyManagement.KeyAgentType.Trezor
+        ? HardwareTrezor.TrezorKeyAgent.checkDeviceConnection(KeyManagement.CommunicationType.Web)
+        : HardwareLedger.LedgerKeyAgent.establishDeviceConnection(Wallet.KeyManagement.CommunicationType.Web));
```
Comment (tomislavhoracek) [inline apps/browser-extension-wallet/src/features/dapp/components/ConfirmTransaction.tsx:173]:
```suggestion
        : HardwareLedger.LedgerKeyAgent.establishDeviceConnection(KeyManagement.CommunicationType.Web));
```

**Review by tomislavhoracek — COMMENTED** [review]
@DominikGuzei 💯 
I found only one minor thing

File: apps/browser-extension-wallet/src/features/dapp/components/ConfirmTransaction.tsx (hunk 1 above)
Comment (DominikGuzei) [reply apps/browser-extension-wallet/src/features/dapp/components/ConfirmTransaction.tsx]: fixed in https://github.com/input-output-hk/lace/pull/862/commits/eadf5a57287f6f02fbdc1c34ac08fa1787cf45c6

**Review by tomislavhoracek — APPROVED** [review]

**Review by AngelCastilloB — APPROVED** [review]
change look good @DominikGuzei 🚀

**Pull request closed without merging**